src/bic/model/application.js

```diff
@@ -45,10 +45,28 @@ define(function (require) {
 
     idAttribute: '_id',
 
+    /**
+     * @property {string} _id - The Answer space id
+     * @property {string} siteName - The site name
+     * @property {string} currentURL - The current URL
+     * @property {string} loginStatus - If the user is logged in or logged out
+     * @property {boolean} displayErrorSummary - controls the error summary display
+     * @property {string} defaultTransition - 'fade'. See http://api.jquerymobile.com/1.3/global-config/
+     * @property {string} recordNotFoundMessage - A Message to display to the user when a requested record is not found on the server.
+     */
     defaults: {
       _id: window.BMP.BIC.siteVars.answerSpace.toLowerCase(),
       loginStatus: false,
-      displayErrorSummary: true
+      displayErrorSummary: true,
+      siteName: undefined,
+      currentURL: undefined,
+
+      homeScreen: undefined,
+      homeInteraction: undefined,
```

Comment:
Those two eslint rules are for different use cases. One is stopping unnecessary characters, the other is to save people from being an idiot and redefining 'undefined'.  Neither of those two apply here. Even the creator of eslint says that many of the rules are unnecessary and only enables a few of them for his projects.

The purpose of adding the the defaults hash with an unitialised value is because no one has any idea exactly what attributes a model has unless they read every file. Obviously this is not the best. 

I dont think that there is anything wrong with saying the properties of the defaults hash are undefined, as they technically are undefined at that point. But, you're right, `null` is probably a  technically better fit for when the property will be filled in at a later time.

I'll also go and change `undefined` to `null` in the other places where the defaults hash defines them as undefined, so we can be consistent.